erpnext/selling/sales_common.js

```diff
@@ -463,24 +463,50 @@ erpnext.selling.SellingController = erpnext.TransactionController.extend({
 	}
 });
 
-frappe.ui.form.on(cur_frm.doctype,"project", function(frm) {
-	if(in_list(["Delivery Note", "Sales Invoice"], frm.doc.doctype)) {
-		if(frm.doc.project) {
-			frappe.call({
-				method:'erpnext.projects.doctype.project.project.get_cost_center_name' ,
-				args: {	project: frm.doc.project	},
-				callback: function(r, rt) {
-					if(!r.exc) {
-						$.each(frm.doc["items"] || [], function(i, row) {
-							if(r.message) {
-								frappe.model.set_value(row.doctype, row.name, "cost_center", r.message);
-								frappe.msgprint(__("Cost Center For Item with Item Code '"+row.item_name+"' has been Changed to "+ r.message));
-							}
-						})
+frappe.ui.form.on(cur_frm.doctype, {
+	project: function(frm) {
+		if(in_list(["Delivery Note", "Sales Invoice"], frm.doc.doctype)) {
+			if(frm.doc.project) {
+				frappe.call({
+					method:'erpnext.projects.doctype.project.project.get_cost_center_name' ,
+					args: {	project: frm.doc.project	},
+					callback: function(r, rt) {
+						if(!r.exc) {
+							$.each(frm.doc["items"] || [], function(i, row) {
+								if(r.message) {
+									frappe.model.set_value(row.doctype, row.name, "cost_center", r.message);
+									frappe.msgprint(__("Cost Center For Item with Item Code '"+row.item_name+"' has been Changed to "+ r.message));
+								}
+							})
+						}
 					}
-				}
-			})
+				})
+			}
 		}
+	},
+
+	order_type: function(frm) {
```

Comment:
This will override any `order_type` calls in individual doctypes. Add a `_super` call to ensure that doesn't happen.

```suggestion
	order_type: function(frm) {
		this._super(frm);
```